shuffle-project/asr-api
# Feature: Add simple Websocket endpoint incl. Testclient using microphone

## What has been done?

closes #25 
closes #37 

### API Websocket Server

This PR introduces a Websocket server to our API, allowing clients to send audio snippets as `bytearray`. The Whisper runner processes these and returns the transcription to the client in the WhisperCPP default JSON.

**Highlights:**
- ResilientWebsocket client, with timeout for unprocessed/erroring audio bytes.
- Settings are from now on passed to the runner. This enables automatic language detection.

### Runner Updates

- Separated the file handling logic.
- New logger helper for better debugging.
- Added **Tiny Model** for enhanced performance.
- Bug fixes and optimizations for faster real-time transcription.
- Added periodic cleanup of old status files.

### Example Scripts

- `example/microphone.py` for testing the Websocket API endpoint.

### Testing Instructions

1. Start with `docker compose -f docker-compose.local.yml up`.
2. Use the tiny model (default in this branch).
3. Test with `example/microphone.py` and your notebook's mic. (make sure that all pip package dependencies are installed, this might not be in the `requirements.txt`)
4. Speak into the mic to see near real-time transcription.
